python/tk_framework_desktopserver/shotgun/api_v2.py

```diff
@@ -621,6 +592,68 @@ def _get_lookup_hash(self, config_uri, project, entity_type):
             entity_type=entity_type
         )
 
+    def _get_pipeline_configuration_data(self, manager, project_entity, data):
+        """
+        Gathers all of the necessary data pertaining to the project's pipeline
+        configurations. This includes the PipelineConfiguration entity, the
+        contents and lookup hashes, and the associated descriptor object.
+
+        :param manager: A ToolkitManager.
+        :param dict project_entity: The Project entity dict.
+        :param dict data: The payload from the client.
+
+        :returns: A dictionary, keyed by PipelineConfiguration entity id, that
+            contains dictionaries with "contents_hash", "lookup_hash",
+            "descriptor", and "entity" keys.
+        :rtype: dict
+        """
+        if "config_data" in self.WSS_KEY_CACHE[self._wss_key]:
+            self.logger.debug("Pipeline config data found for %s", self._wss_key)
+        else:
+            config_data = dict()
+            pcs = self._get_pipeline_configurations(
```

Comment:
Breaking it out into two lines and commenting it so it's easier to read and understand.